SDL Warning prevents compiling

### Describe the bug

Attempting to compile the current build of the game results in:
```

C:/msys64/mingw64/include/SDL2/SDL_main.h:143:17: error: no previous declaration for 'int SDL_main(i
nt, const char**)' [-Werror=missing-declarations]
  143 | #define main    SDL_main
      |                 ^~~~~~~~

```

#70377 sought to fix it but that solution didn't work. I popped into the discord and didn't really see anybody talking about it, so I'm making an issue in case it needs more eyes on it. If it's already under control then please disregard.

### Attach save file

N/A

### Steps to reproduce

Clone the repository, try to compile, fail.

### Expected behavior

The game ought to compile.

### Screenshots

_No response_

### Versions and configuration

If it matters, I'm using MSYS2 and the command I'm trying is $ make CCACHE=0 RELEASE=1 MSYS2=1 DYNAMIC_LINKING=1 SDL=0 TILES=1 SOUND=1 LOCALIZE=1 LANGUAGES=all LINTJSON=0 ASTYLE=0 RUNTESTS=0 -j12

If there's a short-term fix I could do on my end I'd love to hear about it so I can get back to work.

### Additional context

_No response_

--- a/src/sdl_wrappers.h
+++ b/src/sdl_wrappers.h
@@ -2,6 +2,7 @@
 #ifndef CATA_SRC_SDL_WRAPPERS_H
 #define CATA_SRC_SDL_WRAPPERS_H
 
+#define SDL_MAIN_HANDLED
 // IWYU pragma: begin_exports
 #if defined(_MSC_VER) && defined(USE_VCPKG)
 #   include <SDL2/SDL.h>

